app/controllers/editions_controller.rb

```diff
@@ -1,8 +1,19 @@
 # frozen_string_literal: true
 
 class EditionsController < ApplicationController
+  before_action :check_permissions
+
   def create
     Editions::CreateInteractor.call(params: params, user: current_user)
     redirect_to edit_document_path(params[:document])
   end
+
+private
+
+  def check_permissions
+    @edition = Edition.find_current(document: params[:document])
+    return if current_user.can_access_in_history_mode?(@edition) || !@edition.history_mode?
```

Comment:
It seems somewhat unnecessary to have this `can_access_in_history_mode?` method if we also need to check if the edition is in history mode. Might as well be:

```
if @edition.history.mode? && !current_user.has_permission?(User::MANAGE_LIVE_HISTORY_MODE)
  render :non_manage_live_history_mode, status: :forbidden
end
```

Reply:
Removed the method from the user model 👍